src/mscorlib/shared/System/MemoryDebugView.cs

```diff
@@ -0,0 +1,40 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+
+namespace System
+{
+    internal sealed class MemoryDebugView<T>
+    {
+        private readonly ReadOnlyMemory<T> _memory;
+
+        public MemoryDebugView(Memory<T> memory)
+        {
+            _memory = memory;
+        }
+
+        public MemoryDebugView(ReadOnlyMemory<T> memory)
+        {
+            _memory = memory;
+        }
+
+        [DebuggerBrowsable(DebuggerBrowsableState.RootHidden)]
+        public T[] Items
+        {
+            // This is a work around since we cannot use _memory.ToArray() due to
+            // https://devdiv.visualstudio.com/DevDiv/_workitems?id=286592
+            get
+            {
+                if (_memory.DangerousTryGetArray(out ArraySegment<T> segment))
+                {
+                    T[] array = new T[_memory.Length];
+                    Array.Copy(segment.Array, segment.Offset, array, 0, array.Length);
+                    return array;
+                }
+                return Array.Empty<T>();
```

Comment:
Does the debugger issue also prevent us from getting the memory's span and then filling in an array?  It seems problematic that if the memory wrapped something other than an array, we're going to return an empty array even if the memory wasn't empty.  If there's no way to handle that case currently, let's at least make sure we have a coreclr issue tracking this, and include the issue number in the comment in the code.

Reply:
> Does the debugger issue also prevent us from getting the memory's span and then filling in an array?

Yes, accessing the Span property results in debugger view error:
Error = Unable to evaluate the expression. Operation not supported. Unknown error: 0x80070057.

> It seems problematic that if the memory wrapped something other than an array, we're going to return an empty array even if the memory wasn't empty. 

We would return an empty array only if the TryGetArray on OwnedMemory fails:
https://github.com/dotnet/corefx/blob/master/src/System.Memory/src/System/Memory.cs#L228
So, if there is a way to get an array from a memory wrapped in something other than an array, the concrete implementation of OwnedMemory should provide an override of TryGetArray that works.

> let's at least make sure we have a coreclr issue tracking this, and include the issue number in the comment in the code.

Are you suggesting we figure out a way to return the correct array segment even if TryGetArray returns false?
https://github.com/dotnet/corefx/blob/master/src/System.Memory/src/System/Memory.cs#L240

Reply:
> Are you suggesting we figure out a way to return the correct array segment even if TryGetArray returns false?

Yes, as it would for example with https://github.com/dotnet/corefx/blob/master/src/Common/tests/System/Buffers/NativeOwnedMemory.cs.  TryGetArray on such a type should not be allocating a new array and copying into it; the whole point of TryGetArray is to provide a fast path when the OwnedMemory does wrap an array, and allocating / copying into an array is far from the intention of such a fast path.

Reply:
> The purpose of this debugger view is to show the elements of the memory, regardless of what backs it.

That makes sense. If getting a Span from Memory works in the debugger view, I can update the implementation to show the elements correctly. I will keep on eye on the internal bug listed in the comments. We need that bug fixed anyway to get Span debugger view working (for fast span).